### ISSUE TITLE
Fulltext search on item names

### ISSUE THREAD

**Issue by eozalp** [jekkos/opensourcepos#1072]
is there "fulltext search" for item names ? How we can add fulltext search capability ?

**Comment by odiea** [comment]
Yes. There appears to be. In Receivings or Register the item name can be searched. In Items you can search for names,categories and if you select search custom it also works if the item has it.

**Comment by eozalp** [comment]
let's explain more then.

I have a lot of similar products. Because of this i named them with properties.

Example 1 : "round red onyx"
Example 2 : " ovale onyx red"


When i search "onyx red" i just get 2nd example. after a little googling i find that in mysql there is a feature called "fulltext search". I changed item.name column type as "fulltext". But i couldnt locate where i have to change the php code according to this. There must be a line which says  " select names from items match (name ) against (thisquery)" .


With this change maybe i can filter products with properties.

Here comes reference:

http://dev.mysql.com/doc/refman/5.7/en/fulltext-search.html

**Comment by odiea** [comment]
You can look in application/models/items.php and play around with that. I am lazy so I would just insert the keyword onyx.

**Comment by eozalp** [comment]
I played around a little. Where is db->or_like() function? İ couldnt locate.

**Comment by odiea** [comment]
I believe that is built into MySQL.

**Comment by eozalp** [comment]
i know. But i find that too. When you input anything to search bar there is no parsing. İt just go through code as one string including dots commas and space. So you cant search with words. İf you want to find a product you have to remember its whole name or a part of its name. 

İf you have a thousand products in your shop this search bar doesnt help unfortunately. 

when your new products arrived and you have to barcode them and if you cant find your product barcode number. this is the big pain.

**Comment by jekkos** [comment]
I'd recommend to use barcodes anyway. Normally every product should have one by default. This helps a lot in the sense that it will save you some mistakes in the end.

To make search more specific attributes could be used but the feature is not completely finished yet

**Comment by daN4cat** [comment]
Similar discussion in #662

**Comment by eozalp** [comment]
string parse needed in db queries. and in mysql item_name needs index feature. 

it is a "must be" feature. When i found that "onyx red" and "red onyx" gets different results i thought its a glitch in my system. 

This is an open-source application. if only i had coding experience. Here is my complete thougts.

1 - the first thing is search. Trouble explained.
2- there is no "sale stream". you cant see what you sold. i think the system must be more "picture centric". İn sale and receivings the product avatar  can be called from db.

3- in reports product names may have links to the product details. 

Those are my thoughts.

I am using this software and i like it. Just for its sake i am trying to learn php, mysql, mvc, css.

**Comment by eozalp** [comment]
ii made a simple and stupid php file for full text search temporarily. you can use boolean operators.
here is steps:

* go your phpmyadmin and find ospos table. query this : alter table ospos_items add fulltext('name')
*copy the given php file and paste it to your ospos's public folder. https://gist.github.com/eozalp/bafd86fbc291aafdc3dd1ccf108cd2e5
*stop and start.
*call the file from your browser.

**Comment by eozalp** [comment]
i wrote a simple script to search:

https://gist.github.com/eozalp/bafd86fbc291aafdc3dd1ccf108cd2e5
